03-make/make_soup/makefile

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/make
+
+kitchen=./kitchen
+food-market=./food-market
+dinning-room=./dinning-room
+table=${kitchen}/table
+knife=${kitchen}/knife
+pot=${table}/pot
+spoon=${table}/spoon
+
+serve_soup.o: make_soup.o
+	echo
+	echo "Serve some soup"
+	cp ${pot}/soup.txt ${dinning-room}/plate_of_soup.txt
+	echo "Soup served"
+
+make_soup.o: add_ingridients.o  boil_water.o
+	echo
+	echo "Make soup"
+	cat ${pot}/boiled_water.txt > ${pot}/soup.txt
+	rm ${pot}/boiled_water.txt
+	cat ${pot}/cut_carrot.txt >> ${pot}/soup.txt
+	rm ${pot}/cut_carrot.txt
+	cat ${pot}/cut_potato.txt >> ${pot}/soup.txt
+	rm ${pot}/cut_potato.txt
+	cat ${pot}/salt.txt >> ${pot}/soup.txt
+	rm ${pot}/salt.txt
+	echo "Soup done"
+
+boil_water.o:
+	echo
+	echo "boil water"
+	mv ${pot}/water.txt ${pot}/boiled_water.txt
+	echo "some boiled water">${pot}/boiled_water.txt
+	echo "water boiled"
+
+ingridients.o:  get_ingridients.o prepare_ingridients.o
+	echo
+	echo "ingridients for soup are ready"
+
+get_ingridients.o:
+	echo
+	echo "get ingridients"
+	cp -a ${food-market}/. ${table}/
```

Comment:
Are /. and trailing / required?

Reply:
as i read it's a special cp requirement for this case and it was failing without "/."